atlite/gis.py

```diff
@@ -14,30 +14,65 @@
 import numpy as np
 import pandas as pd
 import xarray as xr
-import pickle
-import scipy as sp, scipy.sparse
+import scipy as sp
+import scipy.sparse
 from collections import OrderedDict
 from warnings import warn
-from itertools import product
 from functools import partial
 import pyproj
-from shapely.prepared import prep
+import geopandas as gpd
 from shapely.ops import transform
-from shapely.geometry import box
 import rasterio as rio
 import rasterio.warp
-from rasterio.warp import Resampling
-from rtree.index import Index
 from shapely.strtree import STRtree
 
 import logging
 logger = logging.getLogger(__name__)
 
+
+def get_coords(x, y, time, dx=0.25, dy=0.25, dt='h', **kwargs):
+    """
+    Create an cutout coordinate system on the basis of slices and step sizes.
+
+    Parameters
+    ----------
+    x : slice
+        Numerical slices with lower and upper bound of the x dimension.
+    y : slice
+        Numerical slices with lower and upper bound of the y dimension.
+    time : slice
+        Slice with strings with lower and upper bound of the time dimension.
+    dx : float, optional
+        Step size of the x coordinate. The default is 0.25.
+    dy : float, optional
+        Step size of the y coordinate. The default is 0.25.
+    dt : str, optional
+        Frequency of the time coordinate. The default is 'h'. Valid are all
+        pandas offset aliases.
+
+    Returns
+    -------
+    ds : xarray.Dataset
+        Dataset with x, y and time variables, representing the whole coordinate
+        system.
+    """
+    x = slice(*sorted([x.start, x.stop]))
+    y = slice(*sorted([y.start, y.stop]))
+
+    ds = xr.Dataset({'x': np.arange(-180, 180, dx),
+                     'y': np.arange(-90, 90, dy),
+                     'time': pd.date_range(start="1979", end="now", freq=dt)})
+    ds = ds.assign_coords(lon=ds.coords['x'], lat=ds.coords['y'])
+    ds = ds.sel(x=x, y=y, time=time)
+    return ds
```

Comment:
Should we add the concept of a projection into this `get_coords` function? Think rotpole transform for cordex for example.

I'd propose to have this as a separate issue, though!